src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,116 @@
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectedPostId: number;
+};
+
+export const PostDetails: React.FC<Props> = (props) => {
+  const { selectedPostId } = props;
+  const [details, setDetails] = useState({} as Post);
+  const [comments, setComments] = useState([] as Comment[]);
+  const [commentsDisplay, setCommentsDisplay] = useState(true);
+  const [loading, setLoadingStatus] = useState(true);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(response => {
+        setDetails(response);
+        setLoadingStatus(false);
+      });
+  }, [selectedPostId]);
+
+  useEffect(() => {
+    getPostComments(selectedPostId)
+      .then(response => setComments([...response]));
+  }, [selectedPostId]);
+
+  const handleRemove = (id: number) => {
+    deleteComment(id)
+      .then(() => getPostComments(selectedPostId))
+      .then(updatedComments => setComments(updatedComments));
+  };
+
+  const addComment = (newComment: Comment) => {
+    addCommentToServer(newComment)
+      .then(() => getPostComments(selectedPostId))
+      .then(updatedComments => setComments(updatedComments));
+  };
+
+  const showComments = () => {
+    if (commentsDisplay) {
+      setCommentsDisplay(false);
+    } else {
+      setCommentsDisplay(true);
+    }
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      {loading && (
+        <Loader />
+      )}
+
+      <section className="PostDetails__post">
+        <p>
+          {details.body}
+        </p>
+      </section>
+
+      {comments.length > 0 && (
+        <>
+          <section className="PostDetails__comments">
+            <button
+              type="button"
+              className="button"
+              onClick={() => showComments()}
+            >
+              {commentsDisplay ? (
+                `Hide ${comments.length} comments`
+              ) : (
+                'Show comments'
+              )}
+            </button>
+
+            {commentsDisplay && (
```

Comment:
I think, if you put conditional rendering into `ul`  it would be better (right now your hide/show button don't looks good when you hide comments)

Reply:
Really good point! 👍 Thanks! I've changed that